### PR TITLE
Refactor layout

### PR DESCRIPTION
**Pull request by p3rcypj**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/865cm3qjj

### :memo: Implementation

- Relocate tracks vertically in the center, so they perfectly fit in the track view
- Make tracks taller only when more than 2 fragments sharing same position of the sequence (previous was when more than 1)
- Move track actions on a tooltip (hide when leaves the protvista-pdb frame)
- Change button icons and descriptions
- View and hide tooltips set position as fixed (calculated position) so no longer tooltip cuts
- Set the display of the layout of the tracks as flex, aligned center with fixed width on the labels container
- Toggle highlight and zoom icons (resets on other tracks)
- Removed `labelTooltip`
- Display the name/help on hover as title attribute

### :art: Screenshots

Tracks height **BEFORE**:

image

Tracks height **AFTER**:

image

New layout demo:

3dbio

### CODE REVIEW

**Comment by p3rcypj** [comment]
@tokland I removed `labelTooltip` to support a new info tooltip. The info tooltip only gets the description and the title, but I saw in comments that `labelTooltip` was capable of html. Can it really store html to represent after all? I saw a few lines with that prop being settled with subtrack's label and description, not html.

File: src/section-templates/tracks.js [hunk 1]
```diff
@@ -83,26 +90,14 @@ function getHighlightClass(ctx, trackIndex, subtrackIndex) {
         isHighlighted ? "enabled" : "",
     ]
 
-    return classes.filter(Boolean).join(" ");
+    return classes.filter(Boolean).join(" ");
```
Comment (tokland) [inline src/section-templates/tracks.js:93]: Trailing spaces

File: src/helpers/layout.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ class LayoutHelper {
     constructor(ctx) {
         this.ctx = ctx;
         this.handlers = [];
+        this.ctx.addEventListener('mouseleave',()=>{this.hideMoreOptions();});
```
Comment (tokland) [inline src/helpers/layout.js]: Missing spaces. We don't have a prettifier, but let's use the typical styling guidelines (in accordance with the specific of this project).

File: src/helpers/layout.js [hunk 3]
```diff
@@ -396,13 +397,22 @@ class LayoutHelper {
         if(this.ctx.zoomedTrack != ''){
             let prevZoomIconEle = this.ctx.querySelector('.pvZoomIcon_'+this.ctx.zoomedTrack);
             prevZoomIconEle.classList.remove('active');
+            const icon=prevZoomIconEle.children[0];
+            const text=prevZoomIconEle.children[1];
+            icon.classList.remove("icon-search-minus");
```
Comment (tokland) [inline src/helpers/layout.js]: icon/text may not exist, should check with if

File: src/helpers/layout.js [hunk 4]
```diff
@@ -396,13 +397,22 @@ class LayoutHelper {
...
+            icon.classList.add("icon-search-plus");
+            text.innerText="Zoom in";
```
Comment (tokland) [inline src/helpers/layout.js]: This pattern is repeated 4 times, right? (with different element/classes/text). We can abstract it to a function.

File: src/helpers/layout.js [hunk 5]
```diff
@@ -527,13 +544,63 @@ class LayoutHelper {
...
+    hideMoreOptions(){
+        //Hide all more options menu
+        if(this.ctx)
+            [...this.ctx.querySelectorAll('.moreOptionsMenu')].forEach(m=>m.style.display = 'none');
+    }
+
+    hideInfoTooltips(){
+        //Hide all info tooltips
+        if(this.ctx)
+            [...this.ctx.querySelectorAll('.infoTooltip')].forEach(m=>m.style.display = 'none');
```
Comment (tokland) [inline src/helpers/layout.js:579]: Prettify spaces (for all the new code)

**Review by tokland — COMMENTED** [review]
Some minor comments:

**Review by tokland — APPROVED** [review]
All good, thx!

**Pull request merged**